packages/tinacms/src/toolkit/fields/field-utils.ts

```diff
@@ -0,0 +1,12 @@
+export const hasRTLChar = (text: string): boolean => {
+  return /[\u0600-\u06ef]/g.test(text)
+}
+
+export const getDirection = (
+  value: number | string | boolean | readonly string[]
+) => {
+  if (typeof value === 'string') {
+    return hasRTLChar(value) ? 'rtl' : 'ltr'
+  }
+  return 'unset'
+}
```

Comment:
Can we instead use the browser `Intl.Locale` to do the work for us? 

Something like: 

```
/*
Get the navigator language. Used to Auto format Fields direction based on the navigator language.
*/
export const getNavigatorLanguage = () =>
  navigator.languages && navigator.languages.length
    ? navigator.languages[0]
    : navigator.language || 'en'

/*
Parse the fields.lang to get the direction of the text.
*/
export const getDirection = (lang: string) => {
  const locale = new Intl.Locale(lang)
  //@ts-ignore
  return locale.textInfo.direction
}

```

Then we could add a new field config called `lang` that uses https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/language to select what direction. Off the top of my head, this could have a default set as `en`, then another options as `auto` to use the users Browser default or a specific language you set.